lib/susedistribution.pm

```diff
@@ -473,6 +473,14 @@ sub console_selected {
     # x11 needs special handling because we can not easily know if screen is
     # locked, display manager is waiting for login, etc.
     return ensure_unlocked_desktop if $args{tags} =~ /x11/;
+    # press return twice in console to make sure that PS1 prompt is visible
+    if ($args{tags} =~ /console|shell/) {
+        if (!check_screen($args{tags}, 10)) {
```

Comment:
yes, this might work. I would not even mind the default timeout here because by default the screen *does* match. I am sorry if I annoy you with my "try-to-avoid-check_screen" obsession but actually it only applies in the case when the not-match of the check_screen is the default case. Here the default is the match so we would not waste time. It only waits longer when the terminal does not behave correctly which is not happening that often.

In any case, @rwx788 is already working with me in trying to improve the situation around bsc#1011815, e.g. also see https://progress.opensuse.org/issues/19398